### PR TITLE
Store metadata in crop

### PR DESCRIPTION
**Pull request by Mctalian**
As I mentioned in the other PR:
1. I have no expectations that you will accept this. Feel free to take all of it, some of it, or none of it.
2. I think these changes will make adding new crops in the future a little bit easier by putting as much in Crop and crop-list as possible. This also sets this up to be more data-driven if desired where adding a new crop is adding new data instead of needing to add new code.

Let me know what you think!

### CODE REVIEW

File: components/CropTile.vue [hunk 1]
```diff
@@ -36,32 +36,7 @@ const bonuses = computed(() => {
...
-    case CropType.Blueberry:
-      return 'bg-orange-700'
-    case CropType.Corn:
-      return 'bg-green-800'
-    case CropType.SpicyPepper:
-      return 'bg-amber-600' 
-    default:
-      return ''
-  }
+  return props.tile?.crop?.cropBackgroundColor || ''
```
Comment (Mctalian) [inline components/CropTile.vue:39]: Unfortunately, it appears that the way we're setting Tiles (`this._tiles[row][col].crop = crop`, etc.), Vue doesn't react to that change. I believe we'd need to do something like `this._tiles[row][col] = new Tile(...)` but I'm extremely new to Vue. This is largely what I picked up from the docs.

**Review by Mctalian — COMMENTED** [review]
There's a bug with background color I need to figure out before this is actually ready 🤔 

EDIT: Fixed it.

File: components/CropTile.vue (hunk 1 above)
Comment (Mctalian) [reply components/CropTile.vue:39]: Just kidding, I think it was related to tailwind

**Review by VincentAmante — APPROVED** [review]
This looks good! Definitely makes working with crops much easier in the future

**Comment by VincentAmante** [comment]
Merged as no more issues were found, thanks for your work!

**Pull request merged**